arch/arm64/core/smp.c

```diff
@@ -237,6 +237,6 @@ static int arm64_smp_init(const struct device *dev)
 
 	return 0;
 }
-SYS_INIT(arm64_smp_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+SYS_INIT(arm64_smp_init, PRE_KERNEL_2, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
What does this change have to do with building drivers as a static library?

Reply:
`arm64_smp_init()` has the same init level and priority as the GICv3 interrupt controller. The linking order is affected by the interrupt controller driver being inside a static library, resulting in the interrupt controller being initialized later than `arm64_smp_init()`. There are hard faults running `arm64_smp_init()` as it is setting up IRQs for IPI. This change is simply to make sure the interrupt controller is initialized first.

Reply:
This change is under a separate commit (https://github.com/zephyrproject-rtos/zephyr/pull/37512/commits/1c156742119e983af89b64fde5730e30b8752c30) with explanation in the commit message.

Reply:
The commit message doesn't mention why the link order changed

Reply:
Ah... because that commit is before the changes on driver. I have updated the commit message to say that it is in preparation for the driver changes. Does this work?

Reply:
Yes, thanks